openapi.yaml

```diff
@@ -722,6 +981,131 @@ components:
       additionalProperties: false
       required:
         - permissions
+    Constraints:
```

Comment:
This needs breaking down into "storage constraints" "device constraints" etc.

Reply:
I'm not sure what to do here. The only fields in the `Constraints` that could be included in the `StorageConstraints` (which we already have) are the `root-disk` and the `root-disk-source`, which looking at their description seem awfully related to the device. 

Maybe I can just rename this `Constraints -> DeviceConstraints` or `Constraints -> UnitHardwareConstraints` (which might be more descriptive)? WDYT?